torch/_inductor/decomposition.py

```diff
@@ -229,11 +229,28 @@ def mm(self, input2):
         guard_size_oblivious,
     )
 
-    # Our matrix vector multiplies only achieve peak bandwidth with coordinate descent tuning.
-    # todo: Look into why and fix it (hopefully)
-    if config.coordinate_descent_tuning:
-        if self.shape[0] == 1 or input2.shape[1] == 1:
-            return (self.unsqueeze(2) * input2.unsqueeze(0)).sum(dim=1)
+    def mul_sum_decomp(self, input2):
+        assert self.dtype == input2.dtype
+        out = (self.unsqueeze(2) * input2.unsqueeze(0)).sum(dim=1).to(self.dtype)
+        return out
+
+    def coordesc_or(cond):
+        return config.coordinate_descent_tuning or cond
+
+    if config.decompose_mm_to_mv:
+        gso = guard_size_oblivious
+        # Corresponds to BS=1 cases morally. If it's contiguous we can decompose it
+        # and generate efficient kernels without autotuning. Otherwise we need coordinate descent tuning
+        if definitely_true(self.shape[0] == 1) and coordesc_or(
+            definitely_true(input2.stride(0) == 1)
+        ):
+            return mul_sum_decomp(self, input2)
+
+        if definitely_true(input2.shape[1] == 1) and coordesc_or(
+            definitely_true(self.stride(1) == 1)
```

Comment:
The stride observed during decomposition is not necessarily the stride during lowering. We might want to add a force contiguous node or something on the meta

Reply:
imo, it's fine to leave that up to Inductor. Like, the biggest reason we want to force strides for extern kernels is that Inductor has no visibility into what the extern kernel is going to do, and so if we generate the wrong strides then from Inductor's perspective the kernels may be efficient, but we're actually just pushing a transpose/memory copy *into* the extern kernel.

In this case, if Inductor ended up with a bad layout.... I'd say that's Inductor's problem.

Reply:
I'm looking at a similar issue with bandwidth bound mm decomposition that ends up being extremely slow with non contiguous input. Sure, you can say it's inductor's problem, but it may still lead to regressions today